### ISSUE TITLE
Bug: [no-floating-promises] adds extra parentheses to already-parenthesized expression

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#11486]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&showAST=ts&fileType=.tsx&code=IYZwngdgxgBAZgV2gFwJYHsL3egFAShgG8AoGGXKTEdAGwFMA6W9Ac1wHI4cP8AaGAAUATugC2qEE2H0atAG71O3dL3wBuEgF8SQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTooAd37QmkcGAC%2BIDUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
async function foo() {
  (console.log('foo'), Promise.resolve('foo'));
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-floating-promises": "warn"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

suggestion gives 
```ts
async function foo() {
  await (console.log('foo'), Promise.resolve('foo'));
}
```

### Actual Result

suggestion gives
```ts
async function foo() {
  await ((console.log('foo'), Promise.resolve('foo')));
}
```

### Additional Info

Noticed in https://github.com/typescript-eslint/typescript-eslint/pull/11479 since prettier now formats comma operator expressions as `(a, b);` instead of `a, b;`

**Comment by JoshuaKGoldberg** [comment]
👍 makes sense. This is IMO relatively lower priority as it's a formatting issue. But #11487's changes to the rule are pretty straightforward.
